Realm/RLMSyncSession.mm

```diff
@@ -149,25 +150,24 @@ - (BOOL)waitForDownloadCompletionOnQueue:(dispatch_queue_t)queue callback:(void(
 - (RLMProgressNotificationToken *)addProgressNotificationBlock:(RLMProgressNotificationBlock)block
                                                      direction:(RLMSyncNotifierDirection)direction
                                                           mode:(RLMSyncNotifierMode)mode {
-    return [self addProgressNotificationBlock:block direction:direction mode:mode queue:dispatch_get_main_queue()];
-}
-
-- (RLMProgressNotificationToken *)addProgressNotificationBlock:(RLMProgressNotificationBlock)block
-                                                     direction:(RLMSyncNotifierDirection)direction
-                                                          mode:(RLMSyncNotifierMode)mode
-                                                         queue:(dispatch_queue_t)queue {
     if (auto session = _session.lock()) {
         if (session->state() == SyncSession::PublicState::Error) {
             return nil;
         }
+        // Get the current runloop.
+        CFRunLoopRef currentRunLoop = CFRunLoopGetCurrent();
+        if (!currentRunLoop) {
```

Comment:
I don't think it's possible for `CFRunLoopGetCurrent` to return nil. `CFRunLoopGetCurrent` creates a runloop if one does not already exist.